site/collections/DynamicCollection.tsx

```diff
@@ -0,0 +1,169 @@
+import React from "react"
+import ReactDOM from "react-dom"
+import { BAKED_BASE_URL } from "../../settings/clientSettings.js"
+import { DYNAMIC_COLLECTION_PAGE_CONTAINER_ID } from "@ourworldindata/utils"
+import {
+    IReactionDisposer,
+    ObservableMap,
+    computed,
+    observable,
+    reaction,
+} from "mobx"
+import { observer } from "mobx-react"
+import { WindowGraphers } from "./DynamicCollectionPage.js"
+import { Grapher } from "@ourworldindata/grapher"
+
+interface DynamicCollectionProps {
+    baseUrl: string
+    initialDynamicCollection?: string
+}
+
+/**
+ * After the MultiEmbedder has mounted a Grapher, we poll grapherRef until grapherRef.current is defined,
+ * and then update the window.graphers Map with it.
+ *
+ * This is what allows us to use a reaction in the DynamicCollection component to update the URL whenever a Grapher is updated.
+ */
+export function embedDynamicCollectionGrapher(
+    grapherRef: React.RefObject<Grapher>,
+    figure: Element
+) {
+    const interval = setInterval(() => {
+        if (grapherRef.current) {
+            const originalSlug =
+                grapherRef.current.slug + grapherRef.current.queryStr
+
+            const index = figure.getAttribute("data-grapher-index")
+
+            const windowGrapher = window.graphers.get(
+                `${originalSlug}-${index}`
+            )
+
+            if (windowGrapher) {
+                windowGrapher.grapher = grapherRef.current
+            }
+            clearInterval(interval)
+        }
+    }, 1000)
+}
+
+@observer
+export class DynamicCollection extends React.Component<DynamicCollectionProps> {
+    @observable initialDynamicCollection? = this.props.initialDynamicCollection
+    @observable graphers: undefined | WindowGraphers = undefined
+    pollInterval: null | ReturnType<typeof setInterval> = null
+    disposers: IReactionDisposer[] = []
+
+    @computed get allGrapherSlugsAndQueryStrings() {
+        if (!this.graphers) return []
+
+        // If the grapher hasn't mounted yet, we use the original slugAndQueryString
+        // This allows us to update the URL if users interact with graphers that have mounted
+        // while still keeping the unmounted graphers in the URL in the right place
+        const slugsAndQueryStrings = new Array(this.graphers.size)
+
+        for (const [originalSlugAndUrl, { index, grapher }] of this.graphers) {
+            if (!grapher) {
+                // Strip index suffix from originalSlugAndUrl
+                const withoutIndex = originalSlugAndUrl.replace(/-\d+$/, "")
+                slugsAndQueryStrings[index] = encodeURIComponent(withoutIndex)
+            } else {
+                slugsAndQueryStrings[index] = encodeURIComponent(
```

Comment:
Can we remove `encodeURIComponent` here and above? Grapher URLs should be clean and it would be nice to have a readable URL.